File: app/src/main/java/com/example/mz/udacitypopularmovies/data/MoviesDbHelper.java
package com.example.mz.udacitypopularmovies.data;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

/**
 * Created by mateusz.zak on 07.04.2017.
 */

public class MoviesDbHelper extends SQLiteOpenHelper {
    private static final String DATABASE_NAME = "movies.db";
    private static final int DATABASE_VERSION = 1;

    public MoviesDbHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        final String SQL_CREATE_MOVIES_TABLE = "CREATE TABLE " +
                MovieContract.MovieEntry.TABLE_NAME + " ( " +
                MovieContract._ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                MovieContract.MovieEntry.COLUMN_MOVIE_ID + " INTEGER NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_TITLE + " TEXT NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_OVERVIEW + " TEXT NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_POSTERPATH + " TEXT NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_VOTE_AVERAGE + " REAL NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_RELEASE_DATE + " TIMESTAMP " +
                ");";

        db.execSQL(SQL_CREATE_MOVIES_TABLE);

    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + MovieContract.MovieEntry.TABLE_NAME);
        onCreate(db);
    }
}

File: app/src/main/java/com/mz/popmovies/data/MoviesDbHelper.kt
package com.mz.popmovies.data

import android.content.Context
import android.database.sqlite.SQLiteOpenHelper
import com.mz.popmovies.data.MoviesDbHelper
import android.database.sqlite.SQLiteDatabase

/**
 * Created by mateusz.zak on 07.04.2017.
 */
class MoviesDbHelper(context: Context?) :
    SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        val SQL_CREATE_MOVIES_TABLE = "CREATE TABLE " +
                MovieContract.MovieEntry.TABLE_NAME + " ( " +
                "_id INTEGER PRIMARY KEY AUTOINCREMENT, " +
                MovieContract.MovieEntry.COLUMN_MOVIE_ID.toString() + " INTEGER NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_TITLE.toString() + " TEXT NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_OVERVIEW.toString() + " TEXT NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_POSTERPATH.toString() + " TEXT NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_VOTE_AVERAGE.toString() + " REAL NOT NULL, " +
                MovieContract.MovieEntry.COLUMN_RELEASE_DATE.toString() + " TIMESTAMP " +
                ");"
        db.execSQL(SQL_CREATE_MOVIES_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS " + MovieContract.MovieEntry.TABLE_NAME)
        onCreate(db)
    }

    companion object {
        private const val DATABASE_NAME = "movies.db"
        private const val DATABASE_VERSION = 1
    }
}